src/main/java/net/atos/zac/documenten/InboxDocumentenService.java

```diff
@@ -112,4 +100,45 @@ public void delete(final UUID zaakinformatieobjectUUID) {
             entityManager.remove(inboxDocument);
         }
     }
+
+    public List<InboxDocument> list(final InboxDocumentListParameters listParameters) {
+        final CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+        final CriteriaQuery<InboxDocument> query = builder.createQuery(InboxDocument.class);
+        final Root<InboxDocument> root = query.from(InboxDocument.class);
+        if (listParameters.getSorting() != null) {
+            if (listParameters.getSorting().getDirection() == SorteerRichting.ASCENDING) {
+                query.orderBy(builder.asc(root.get(listParameters.getSorting().getField())));
+            } else {
+                query.orderBy(builder.desc(root.get(listParameters.getSorting().getField())));
+            }
+        }
+        query.where(getWhere(listParameters, root));
+        final TypedQuery<InboxDocument> emQuery = entityManager.createQuery(query);
+        if (listParameters.getPaging() != null) {
+            emQuery.setFirstResult(listParameters.getPaging().getFirstResult());
+            emQuery.setMaxResults(listParameters.getPaging().getMaxResults());
+        }
+        return emQuery.getResultList();
+    }
+
+    private Predicate getWhere(final InboxDocumentListParameters listParameters, final Root<InboxDocument> root) {
+        final CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+        final List<Predicate> predicates = new ArrayList<>();
+        if (StringUtils.isNotBlank(listParameters.getIdentificatie())) {
+            predicates.add(builder.like(root.get("enkelvoudiginformatieobjectID"), LIKE.formatted(listParameters.getIdentificatie())));
```

Comment:
Maak van die literal strings zoals "enkelvoudiginformatieobjectID" en "creatiedatum" een public static final String op InboxDocument Zoals ook bij ZaakafhandelParameters Entity. Helaas doen we dat nog niet overal want het maakt de code onderhoudbaarder.

Reply:
aangepast